### PR TITLE
Security

### PR DESCRIPTION
**Pull request by macben888**
add security package with functionality to login/signup users

### CODE REVIEW

File: backend/src/main/java/capstone/backend/security/service/UserAuthUtils.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package capstone.backend.security.service;
+
+import capstone.backend.security.exceptions.InvalidCredentialsException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserAuthUtils {
+    //TODO implement better password validation
```
Comment (Flooooooooooorian) [inline backend/src/main/java/capstone/backend/security/service/UserAuthUtils.java:8]: remove TODO from code

File: backend/src/main/java/capstone/backend/security/service/UserMapper.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package capstone.backend.security.service;
+
+import capstone.backend.security.model.AppUser;
+import capstone.backend.security.model.UserDTO;
+
+public class UserMapper {
+    public AppUser mapUser(UserDTO user) {
```
Comment (Flooooooooooorian) [inline backend/src/main/java/capstone/backend/security/service/UserMapper.java:7]:
```suggestion
    public static AppUser mapUser(UserDTO user) {
```

File: backend/src/main/java/capstone/backend/security/service/UserMapper.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        return AppUser
+                .builder()
+                .username(user.getUsername())
+                .password(user.getPassword())
+                .build();
+    }
+public UserDTO mapUser(AppUser user) {
```
Comment (Flooooooooooorian) [inline backend/src/main/java/capstone/backend/security/service/UserMapper.java:14]:
```suggestion
public static UserDTO mapUser(AppUser user) {
```

File: backend/src/main/java/capstone/backend/security/config/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        return super.authenticationManagerBean();
+    }
+
+
+    @Override
+    @SuppressWarnings({"java:S5344"})
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/auth/login").permitAll()
```
Comment (Flooooooooooorian) [inline backend/src/main/java/capstone/backend/security/config/SecurityConfig.java:54]: Line is unnecessary

```suggestion
```

File: backend/src/main/java/capstone/backend/security/service/UserAuthService.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.HashMap;
+
+
+@Service
+public class UserAuthService implements UserDetailsService {
+
+    private final UserRepository repository;
+    private final JWTUtilService jwtService;
+    private final PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+    private final UserMapper mapper = new UserMapper();
```
Comment (Flooooooooooorian) [inline backend/src/main/java/capstone/backend/security/service/UserAuthService.java:26]:
```suggestion
```

File: backend/src/main/java/capstone/backend/security/service/UserAuthService.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                        .authorities("user")
+                        .build())
+                .orElseThrow(() -> new UsernameNotFoundException("Username does not exist: " + username));
+    }
+
+    public String signup(UserDTO user) throws InvalidCredentialsException, IllegalArgumentException {
+        validateUserData(user);
+        if (repository.findById(user.getUsername()).isEmpty()) {
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
+            repository.save(mapper.mapUser(user));
```
Comment (Flooooooooooorian) [inline backend/src/main/java/capstone/backend/security/service/UserAuthService.java:52]: Call static UserMapper

File: .github/workflows/sonar-backend.yml [hunk 7]
```diff
@@ -1,5 +1,6 @@
 name: sonar-backend
 on:
+  workflow_dispatch: {}
```
Comment (Flooooooooooorian) [inline .github/workflows/sonar-backend.yml:3]: Why not on push?

File: .github/workflows/sonar-frontend.yml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 name: sonar-fronted
 on:
   push:
-    
+  workflow_dispatch: {}
```
Comment (Flooooooooooorian) [inline .github/workflows/sonar-frontend.yml:4]: Why not on push?

**Review by Flooooooooooorian — APPROVED** [review]

**Comment by Flooooooooooorian** [comment]
Fix Sonar Issues

File: backend/src/test/java/capstone/backend/security/controller/AuthControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+import static org.mockito.Mockito.mock;
+
+@Testcontainers
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class AuthControllerTest {
+
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+    private final RestTemplate restTemplateMock = mock(RestTemplate.class);
```
Comment (christophersiem) [inline backend/src/test/java/capstone/backend/security/controller/AuthControllerTest.java:44]: think you don't need this mock

```suggestion

```

**Pull request merged**